producer.go

```diff
@@ -334,6 +334,8 @@ func (p *Producer) validateMsg(msg any) ([]byte, error) {
 		switch msg.(type) {
 		case []byte:
 			return msg.([]byte), nil
+		case map[string]interface{}:
+			return json.Marshal(msg)
```

Comment:
Fix the comment above the if statement to include the new type. 

Also noticed that you are sending schema failed message inside this if block although it is not schema failure